apps/vault/contracts/Vault.sol

```diff
@@ -1,118 +1,103 @@
 pragma solidity 0.4.18;
 
-import "./VaultBase.sol"; // split made to avoid circular import
+import "@aragon/os/contracts/apps/AragonApp.sol";
 
-import "./connectors/ERC20Connector.sol";
-import "./connectors/ETHConnector.sol";
+import "@aragon/os/contracts/lib/zeppelin/token/ERC20.sol";
 
-import "@aragon/os/contracts/lib/misc/Migrations.sol";
 
+contract Vault is AragonApp {
+    bytes32 constant public TRANSFER_ROLE = keccak256("TRANSFER_ROLE");
 
-contract Vault is VaultBase {
-    struct TokenStandard {
-        uint32 erc;
-        uint32 interfaceDetectionERC;
-        bytes4 interfaceID;
-        address connector;
-    }
-
-    TokenStandard[] public standards;
-    mapping (address => address) public connectors;
-    mapping (uint32 => bool) public supportedInterfaceDetectionERCs;
-
-    // this is required to avoid having to redeploy on every instance.
-    // they can just be gotten from the base by templates
-    ERC20Connector public erc20ConnectorBase;
-    ETHConnector public ethConnectorBase;
-
-    event NewTokenStandard(uint32 indexed erc, uint32 indexed interfaceDetectionERC, bytes4 indexed interfaceID, address connector);
+    event Transfer(address indexed token, address indexed to, uint256 amount);
+    event Deposit(address indexed token, address indexed sender, uint256 amount);
 
-    function Vault() public {
-        erc20ConnectorBase = new ERC20Connector();
-        ethConnectorBase = new ETHConnector();
+    /**
+    * @dev When used behind an AppProxy, this fallback is never executed
+    *      as it will be intercepted by the Proxy (see aragonOS#281)
```

Comment:
Isn't it still possible to trigger this fallback by sending > 10k gas but no data in a call to the proxy? We should probably reword to `On a normal send() or transfer() this fallback is never executed as it will be intercepted by the Proxy (see aragonOS#281)`

Reply:
Actually, should we allow to execute the fallback if it's not behind a proxy? :thinking: What would be the use case for it?

Reply:
I think it's still valuable to have the fallback, in the event someone does use the vault without a proxy. Without it, `send()` and `transfer()` wouldn't work.

Reply:
So, is our Vault supposed to be intended for use without a proxy? This way, couldn't it happen that funds are sent to the base Vault (in an scenario where a proxy is being used) and then are lost?

Reply:
> So, is our Vault supposed to be intended for use without a proxy?

I'd say it's "intended" in a very light way; nothing should break if it's not behind a proxy (things only break if it, or a proxy, is not attached to a kernel).

> couldn't it happen that funds are sent to the base Vault (in an scenario where a proxy is being used) and then are lost

Potentially, if someone got the wrong address for their vault, somehow. We could protect against this by adding an initialization step to the vault, where it only accepts deposits and transfers if it's been initialized.

Reply:
I like this last option, although I still don't see the point of using a proxy-less Vault, plus the fact (as @izqui said) that anybody can initialize it. Maybe we could do a special initialization for base Vault that forbids deposits, although I think it's unnecessary because as I said before I don't get the use case.